tests/averaging/translation/test_averaging_1dsfm.py

```diff
@@ -16,21 +17,99 @@
+            [(C(i2), C(i1)) for (i1, i2) in w_i2Ui1_dict.keys()]
+            + [(C(i2), L(i1)) for (i1, i2) in w_i2Ui1_dict_tracks.keys()]
+        )
+        i1Ui2_measurements = self.obj._binary_measurements_from_dict(w_i2Ui1_dict, w_i2Ui1_dict_tracks, noise_model)
+        actual_measurement_idxs = set([(i1Ui2.key1(), i1Ui2.key2()) for i1Ui2 in i1Ui2_measurements])
+
+        self.assertEqual(expected_measurement_idxs, actual_measurement_idxs)
+
+    def test_binary_measurements_from_priors(self):
+        """Tests that the binary measurements are correctly extracted from priors dict."""
+
+        test_prior = PosePrior(
+            value=Pose3(Rot3(), np.array([1, 1, 0])), covariance=np.eye(6), type=PosePriorType.SOFT_CONSTRAINT
+        )
+        test_wRi_mat = np.eye(3)
+        test_wRi_mat[1, 1] = -1
+        wRi_list = [Rot3(test_wRi_mat)] * 4
+        i2Ti1_dict = {(0, 1): test_prior, (0, 3): test_prior, (1, 2): test_prior}
+
+        expected_measurement_idxs = set([(C(i2), C(i1)) for (i1, i2) in i2Ti1_dict.keys()])
+        w_i1ti2_measurements = self.obj._binary_measurements_from_priors(i2Ti1_dict, wRi_list)
+        actual_measurement_idxs = set([(i1ti2.key1(), i1ti2.key2()) for i1ti2 in w_i1ti2_measurements])
+        self.assertEqual(expected_measurement_idxs, actual_measurement_idxs)
+
+        actual_w_i1ti2 = np.array([i1ti2.measured() for i1ti2 in w_i1ti2_measurements])
+        expected_w_i1ti2 = np.array([Point3(1, -1, 0) for (i1, i2) in i2Ti1_dict.keys()])
+
+        self.assertTrue(np.array_equal(actual_w_i1ti2, expected_w_i1ti2))
+
+    def test_select_tracks_for_averaging(self) -> None:
+        """Tests that the tracks are correctly selected to include in averaging."""
+        valid_cameras = set([0, 1, 2, 3])
+        test_measurement = np.array([20, 20])
+        test_tracks_cameras = {0: [0, 1, 2, 3, 5, 6], 1: [1, 2, 3], 2: [0, 1, 3, 4], 3: [0, 2]}
+        test_tracks = []
+        for track_idx, cameras in test_tracks_cameras.items():
+            measurements = [SfmMeasurement(i, test_measurement) for i in cameras]
+            test_tracks.append(SfmTrack2d(measurements=measurements))
+
+        intrinsics = [Cal3_S2(20, 20, 0.0, 0, 0)] * 7
+        expected_selected_tracks = {0: [0, 1, 2, 3], 1: [1, 2, 3], 2: [0, 1, 3]}
+        actual_selected_tracks = self.obj._select_tracks_for_averaging(test_tracks, valid_cameras, intrinsics)
+        actual_selected_track_cameras = {
+            track_idx: [m.i for m in track.measurements] for track_idx, track in enumerate(actual_selected_tracks)
+        }
+        self.assertEqual(expected_selected_tracks, actual_selected_track_cameras)
+
+    def test_get_landmark_directions(self) -> None:
+        test_measurement = np.array([20, 20])
+        tracks = [SfmTrack2d(measurements=[SfmMeasurement(0, test_measurement)])]
+
+        intrinsics = [Cal3_S2(20, 20, 0.0, 0, 0)]
+        test_wRi_mat = np.eye(3)
+        test_wRi_mat[1, 1] = -1
```

Comment:
Could we add an inline comment explaining what's going on with the euler angles here?